actions/product_descriptions_actions.py

```diff
@@ -0,0 +1,93 @@
+
+from rasa_sdk import Action, Tracker, FormValidationAction
+from rasa_sdk.executor import CollectingDispatcher
+from rasa_sdk.events import (
+    ActionExecuted,
+    SlotSet,
+    UserUtteranceReverted,
+    ActionReverted,
+)
+from rasa_sdk.types import DomainDict
+
+from actions.slot_match import FuzzySlotMatch, FuzzySlotMatchOption, resolve_slot_match
+from actions.platform.chrome import create_service_options
+
+from common import logging
+
+logger = logging.initialize_logging()
+
+
+class ProductDescription(Action):
+    def name(self) -> Text:
+        return "action_product_description"
+
+    async def run(
+        self,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> List[Dict[Text, Any]]:
+        product = next(tracker.get_latest_entity_values("services_generated"), None)
+        print(product)
+
+        # remove console.redhat.com/ from the product name
+        product = product.replace("console.redhat.com", "")
+
+        if product:
+            options = await create_service_options(tracker)
+            if not options:
+                dispatcher.utter_message(
+                    response="utter_product_description_error",
+                    product=product,
+                )
+                return
+            fuzzy_options = [
+                *map(
+                    lambda o: FuzzySlotMatchOption(o["data"]["title"], o["synonyms"]),
+                    options.values(),
+                )
+            ]
+            match = FuzzySlotMatch("slot", fuzzy_options)
+            resolved = resolve_slot_match(product, match)
+
+            if len(resolved) > 0:
+                if (
+                    resolved["slot"] not in options
+                    or "data" not in options[resolved["slot"]]
+                ):
+                    dispatcher.utter_message(
+                        response="utter_product_description_error",
+                        product=product,
+                    )
+                    return
+                details = options[resolved["slot"]]["data"]
+
+                if "title" not in details or "description" not in details:
+                    dispatcher.utter_message(
+                        response="utter_product_description_error",
+                        product=product,
+                    )
+                    return
+                dispatcher.utter_message(
+                    response="utter_product_description",
+                    product=details["title"],
+                    description=details["description"],
+                )
+
+                if "href" in details:
+                    dispatcher.utter_message(
+                        response="utter_product_description_more",
+                        product=details["title"],
+                        href=details["href"],
+                    )
+            else:
+                # can I send the bot to the next likelist intent?
```

Comment:
Did we have a way to get the bot to trigger the other possible intents?

There is no description for top level services (like openshift, ansible) so I'd like it to fallback well.

Reply:
Not sure, but maybe you can get the other possible intents and return a UserUttered (I think that's what it was called).

An alternative would be to have this as part of the pipeline :grin:

Reply:
Would it be possible to remove services that don't have any description?

Reply:
I was going to do that, but I think  it would interrupt a bunch of our flows where a user says something that isnt in the nlu but is fuzzy matched for example

so if I said malware in response to the favoriting flow, how would a descriptions pipeline know not to take over

Reply:
> Would it be possible to remove services that don't have any description?

Oh yeah, ill give that a go

Reply:
I think that the pipeline gets the context (i.e. ability to check the current story, previous intents, utterances, etc)